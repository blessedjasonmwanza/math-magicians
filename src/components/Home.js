const Home = () => (
  <section className="article">
    <h1>
      Welcome to our Page
    </h1>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit.
      Praesentium officia fugit cupiditate itaque autem obcaecati,
      mollitia quis ab soluta, odit in illum rerum nisi nulla adipisci
      incidunt assumenda? Iure, qui?
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit.
      Praesentium officia fugit cupiditate itaque autem obcaecati,
      mollitia quis ab soluta, odit in illum rerum nisi nulla adipisci
      incidunt assumenda? Iure, qui?
    </p>
  </section>
);

export default Home;
