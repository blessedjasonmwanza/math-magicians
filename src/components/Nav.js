import { Link } from 'react-router-dom';
import './css/nav.css';

const Nav = () => (
  <nav className="d-flex rows">
    <h1>Math Magicians</h1>
    <ul className="d-flex">
      <li><Link to="/home">Home</Link></li>
      <li><Link to="/calculator">Calculator</Link></li>
      <li><Link to="/quote">Quote</Link></li>
    </ul>
  </nav>
);

export default Nav;
