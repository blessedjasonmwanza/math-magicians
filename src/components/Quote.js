const Quote = () => (
  <q>
    Don&apos;t stop dreaming and applying your imagination.
    This world is a creation of people who blured out what
    everyone was scared of. Either you want it or not,
    this principal will forever hold true. -
    <b>
      <a href="https://github.com/blessedjasonmwanza" target="_blank" rel="noreferrer">Blessed Jason Mwanza</a>
    </b>
  </q>
);

export default Quote;
