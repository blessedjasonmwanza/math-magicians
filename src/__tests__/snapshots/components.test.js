import React from 'react';
import renderer from 'react-test-renderer';
import {
  Route, BrowserRouter as Router, Routes,
} from 'react-router-dom';
import Calculator from '.../../../components/calculator';
import Home from '.../../../components/Home';
import Nav from '.../../../components/Nav';
import Quote from '.../../../components/Quote';

describe('run screenshots test for each component', () => {
  test('Nav Component renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Nav />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Routes Work correctly', () => {
    const tree = renderer.create(
      <Router>
        <Routes>
          <Route exact path="/" element={<Home />} />
          <Route path="/calculator" element={<Calculator />} />
          <Route path="/Quote" element={<Quote />} />
        </Routes>
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
