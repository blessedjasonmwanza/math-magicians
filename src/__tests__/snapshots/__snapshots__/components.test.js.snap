// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run screenshots test for each component Nav Component renders correctly 1`] = `
<nav
  className="d-flex rows"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="d-flex"
  >
    <li>
      <a
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`run screenshots test for each component Routes Work correctly 1`] = `
<section
  className="article"
>
  <h1>
    Welcome to our Page
  </h1>
  <p>
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Praesentium officia fugit cupiditate itaque autem obcaecati, mollitia quis ab soluta, odit in illum rerum nisi nulla adipisci incidunt assumenda? Iure, qui?
  </p>
  <p>
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Praesentium officia fugit cupiditate itaque autem obcaecati, mollitia quis ab soluta, odit in illum rerum nisi nulla adipisci incidunt assumenda? Iure, qui?
  </p>
</section>
`;
